### PR TITLE
feat(content-security-policies): add nonce and trusted types support

### PR DESCRIPTION
**Pull request by lsagetlethias**
- add nonce support except for Next.js Pages Dir
- add trusted types support
- linked doc change request : https://react-dsfr.codegouv.studio/~/changes/qQlH6EWaIiyp39WwEyKF/content-security-policy

### CODE REVIEW

File: src/next-pagesdir.tsx [hunk 1]
```diff
@@ -177,10 +195,13 @@ export function createNextDsfrIntegrationApi(
                                 />
                             </>
                         )}
-                        {isProduction && (
+                        {isProduction && !isBrowser && (
```
Comment (garronej) [inline src/next-pagesdir.tsx:198]: I guess this is for concistency with App router?

File: src/next-appdir/zz_internal/start.ts [hunk 2]
```diff
@@ -14,8 +16,39 @@ export function startReactDsfr(params: {
...
+     * @default "react-dsfr"
+     */
+    trustedTypesPolicyName?: string;
 }) {
-    const { defaultColorScheme, verbose = false, Link } = params;
+    const {
+        defaultColorScheme,
+        verbose = false,
+        Link,
+        checkNonce,
```
Comment (garronej) [inline src/next-appdir/zz_internal/start.ts:49]:
```suggestion
        checkNonce = false,
```  
Explicitely state the default value

File: src/next-appdir/zz_internal/start.ts [hunk 3]
```diff
@@ -14,8 +16,39 @@ export function startReactDsfr(params: {
     verbose?: boolean;
     /** Default: <a /> */
     Link?: (props: RegisteredLinkProps & { children: ReactNode }) => ReturnType<React.FC>;
+    /**
+     * When true, the nonce of the script tag will be checked, fetched from {@link DsfrHead} component and injected in react-dsfr scripts.
+     *
+     * @see https://developer.mozilla.org/fr/docs/Web/HTML/Global_attributes/nonce
+     */
+    checkNonce?: boolean;
```
Comment (garronej) [inline src/next-appdir/zz_internal/start.ts:24]: Would you consider performing a search and replace for doCheckNonce?
I don't intend to be overly nitpicky, but I aim to enforce a strict naming convention for boolean variables. The convention uses names that clearly indicate questions, such as isXxx, doesXxx, areXxx, etc.  

checkNonce could serve as the name of a function responsible for performing nonce verification.

There is an exception to this naming rule for component props. For instance, <Button disabled /> is more readable than <Button isDisabled />.

**Review by garronej — CHANGES_REQUESTED** [review]
A few details.  
Looks very good to me otherwise.  

Thanks a lot for your work

File: src/next-pagesdir.tsx (hunk 1 above)
Comment (lsagetlethias) [reply src/next-pagesdir.tsx:198]: Yes exactly!

**Comment by garronej** [comment]
Thanks a lot for the changes. I'll merge later today

**Comment by lsagetlethias** [comment]
> Thanks a lot for the changes. I'll merge later today

Thanks! Also, I tagged you as reviewer on the documentation changes on GitBook

**Comment by garronej** [comment]
Okay, there were a few things to tweak:

- \`${undefined} === "undefined"\` (and not `""`).
- Defaults were propagated only at the external API level. Internal functions should have all parameters set as required. This ensures we don't overlook anything and apply defaults only once, universally.
- The `tools` directory is intended for self-contained utilities that solve generic problems. Not a `misc` dir.  

Other than that, everything looks good. Thank you again for your diligent work.

**Pull request merged**